### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by NicholasBatesNZ**
Adds some clarity and new exception throws

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/destinationsol/files/HullConfigManager.java [hunk 1]
```diff
@@ -105,6 +106,8 @@ private HullConfig read(String shipName) {
             readProperties(json.getJsonValue(), configData);
         } catch (IllegalArgumentException e) {
             throw new IllegalArgumentException("The JSON of ship " + shipName + " is missing, or has malformed, a required parameter" + e.getMessage().split(":")[1]);
+        } catch (SerializationException e) {
+            throw new SerializationException("The JSON of ship " + shipName + " has invalid syntax" + e.getMessage().split(":")[1]);
```
Comment (Steampunkery) [inline engine/src/main/java/org/destinationsol/files/HullConfigManager.java]: Take a look at what the error message you're splitting is :wink: 

(hint: it's the entire JSON file)

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**